### PR TITLE
Adds NineSliceSprite and optimizes gallery front page

### PR DESCRIPTION
**Pull request by vlidholt**

### CODE REVIEW

File: packages/flutter_sprites/lib/src/nine_slice_sprite.dart [hunk 1]
```diff
@@ -0,0 +1,191 @@
+part of flutter_sprites;
```
Comment (Hixie) [inline packages/flutter_sprites/lib/src/nine_slice_sprite.dart]: see #1289, by the way

File: packages/flutter_sprites/lib/src/nine_slice_sprite.dart [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+/// A NineSliceSprite is similar to a [Sprite], but it it can strech its
+/// inner area to fit the size of the [Node]. This is ideal for fast drawing
+/// of things like buttons.
+class NineSliceSprite extends NodeWithSize with SpritePaint {
+
+  /// Creates a new NineSliceSprite from the privided [texture], [size], and
```
Comment (Hixie) [inline packages/flutter_sprites/lib/src/nine_slice_sprite.dart]: typo (provided)

File: packages/flutter_sprites/lib/src/nine_slice_sprite.dart [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      VertexMode.triangles,
+      _vertices,
+      _textureCoordinates,
+      _colors,
+      TransferMode.modulate,
+      _indicies,
+      _cachedPaint
+    );
+  }
+}
```
Comment (Hixie) [inline packages/flutter_sprites/lib/src/nine_slice_sprite.dart]: is there some way this class could be tested?

see also #1771.

**Comment by Hixie** [comment]
LGTM with a test for the new sprites class, if that's possible.

**Comment by vlidholt** [comment]
As per the issues you mentioned, I think adding tests and removing the 'part of' will be a part of a larger effort than this particular pull request. Depending on the state of the gallery, I want to start working on this asap.

**Comment by Hixie** [comment]
Removing 'part of' is definitely a separate issue, but it's never too early to start including tests. :-)

**Pull request merged**